xarray/core/dataarray.py

```diff
@@ -6670,6 +6670,68 @@ def resample(
             **indexer_kwargs,
         )
 
+    def roundStringify(self, n: int) -> DataArray:
```

Comment:
we use snake-case for our functions, so it should be `round_stringify`. However, before you change that I agree with @headtr1ck that the method combines two different things (rounding plus string interpolation).

As an API, I would probably prefer something like this:
```python
(
    arr.round(0.01)  # round to two decimal digits
    .str.format("%.03f")  # display three decimals
)
```
where `.str.format` can be implemented in a efficient manner using `numpy.char.mod`.

The advantage of that is that it is a lot more versatile, for example:
```python
ds = xr.Dataset({"a": ("x", ["a", "b", "c"]), "b": ("x", [0.2756, 6.7279, 3.7832])})
ds.a.str.format("%s: ").str + ds.b.str.format("%.03")
```

Reply:
> @keewis [str.format](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.str.format.html) already exists and behaves like the python str.format method (also different from the builtin [`format()`](https://docs.python.org/3/library/functions.html#format) method).
> 

Good find.
This method is working the other way around as the proposed solution, but it should work perfectly.

Untested:
```python

fmt = xr.DataArray("{:.2f}")
da = xr.DataArray([1, 2, 3.456], dims="x")

fmt.str.format(da)
```

Reply:
the trouble with that is that it will "stringify" the values, so the only format you can pass it is `:s`. We should probably change that to leave these assumptions to the actual string formatting, though.